app/controllers/distributions_controller.rb

```diff
@@ -86,26 +87,27 @@ def new
   end
 
   def show
-    @distribution = Distribution.includes(:line_items).includes(:storage_location).find(params[:id])
     @line_items = @distribution.line_items
   end
 
   def edit
-    @distribution = Distribution.includes(:line_items).includes(:storage_location).find(params[:id])
-    @distribution.line_items.build
-    @items = current_organization.items.alphabetized
-    @storage_locations = current_organization.storage_locations.alphabetized
+    if distribution.can_be_edited? || current_user.organization_admin?
+      @distribution.line_items.build
+      @items = current_organization.items.alphabetized
+      @storage_locations = current_organization.storage_locations.alphabetized
+    else
+      flash[:error] = 'To edit a distribution,
+      you must be an organization admin or the current date must be major then today.'
+      redirect_to distributions_path
+    end
   end
 
   def update
-    distribution = Distribution.includes(:line_items).includes(:storage_location).find(params[:id])
-
     # there are ways to convert issued_at(*i) to Date but they are uglier then just remember it here
     # see examples: https://stackoverflow.com/questions/13605598/how-to-get-a-date-from-date-select-or-select-date-in-rails
     old_issued_at = distribution.issued_at
 
     if distribution.replace_distribution!(distribution_params)
-      @distribution = Distribution.includes(:line_items).includes(:storage_location).find(params[:id])
```

Comment:
I think this will create a bug here - under the proposed way @distribution will be cached as the old distribution, but we want to set it to the new distribution.

Reply:
Resolved